eduNEXT/frontend-app-validation-panel
# feat: create the storage to manage the app state

**Create Data Storage**
This PR creates the storage to manage the app state and the connection with the backend.

The functions were created considering the technical documentation but could be modified in the future according to any API specification.

**How to test**
1. Modify the ValidationPanelPage to display the information in the browser
```
import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { getCourses } from './data/thunks';

const ValidationPanelPage = () => {
  const { courses } = useSelector(state => state.courses);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getCourses());
  }, [dispatch]);

  return (
    <div>{courses.map(({ courseId }) => <h1>{courseId}</h1>)}</div>
  );
};

export default ValidationPanelPage;
```
2. You should see the course ID for each course visible to the current user. 

Note: this exmple use the course API http://<your_lms>:8000/api/courses/v1/courses/?username=<current_user>
[JIRA CARD](https://edunext.atlassian.net/browse/DS-634)

You can use a mock api rest to test the validation API.
